UKHomeOffice/sgmr-service
# SPCR-490 Refactor repeated/untidy code

### AC / Description of changes
Some of the code is not DRY and some if-else statements are long and untidy - refactor the code so that these areas are cleaned up

- Error summary
- Arrival and departure (handleSubmit function)

### To Test
- Create a voyage
- Click save and **continue**
-  > All fields should error as expected and scroll to and focus correctly
- Fill in correct details
- Click save and **continue**
- > You should be taken to the arrival page
- Go back and make a change
- Click save and **exit**
- Return to the draft voyage
- > Your changes should persist
- Return to the arrival page
- It should error correctly as with the departure page

### Notes

---

**Remember to merge to feature branches not `master`.**
